### PR TITLE
Migrate hot reload tests to Dart 2

### PR DESCRIPTION
**Pull request by cbracken**

### CODE REVIEW

File: packages/flutter_tools/test/hot_test.dart [hunk 1]
```diff
@@ -91,34 +93,41 @@ void main() {
...
+      final List<FlutterDevice> devices = <FlutterDevice>[new FlutterDevice(new MockDevice(), generator: residentCompiler, trackWidgetCreation: false, previewDart2: true)];
+      expect((await new HotRunner(devices).restart(fullRestart: true)).isOk, true);
+    }, overrides: <Type, Generator>{
+      Artifacts: () => mockArtifacts,
     });
 
     testUsingContext('setup function succeeds', () async {
-      expect((await new HotRunner(devices).restart(fullRestart: true)).isOk,
-          true);
+      final List<FlutterDevice> devices = <FlutterDevice>[new FlutterDevice(new MockDevice(), generator: residentCompiler, trackWidgetCreation: false, previewDart2: true)];
```
Comment (goderbauer) [inline packages/flutter_tools/test/hot_test.dart]: nit: These lines may be rip for a line break.

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter_tools/test/hot_test.dart (hunk 1 above)
Comment (cbracken) [reply packages/flutter_tools/test/hot_test.dart]: I'm about to followup with another patch that deletes the previewDart2 parameter, which'll get it just barely short-enough that it'll fit reasonably on one slightly long line.

**Comment by cbracken** [comment]
Partially addresses #17051 and #17804.

**Pull request merged**